File: mifosng-android/src/main/java/com/mifos/mifosxdroid/offlinejobs/OfflineJobCreator.java
package com.mifos.mifosxdroid.offlinejobs;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.evernote.android.job.Job;
import com.evernote.android.job.JobCreator;
import com.mifos.utils.Tags;

/**
 * Created by aksh on 17/7/18.
 */

public class OfflineJobCreator implements JobCreator {
    @Nullable
    @Override
    public Job create(@NonNull String tag) {
        switch (tag) {
            case Tags.OfflineSyncCenter:
                return new OfflineSyncCenter();
            case Tags.OfflineSyncClient:
                return new OfflineSyncClient();
            case Tags.OfflineSyncGroup:
                return new OfflineSyncGroup();
            case Tags.OfflineSyncSavingsAccount:
                return new OfflineSyncSavingsAccount();
            case Tags.OfflineSyncLoanRepayment:
                return new OfflineSyncLoanRepayment();
            default:
                return null;
        }
    }
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/offlinejobs/OfflineJobCreator.kt
package com.mifos.mifosxdroid.offlinejobs

import com.evernote.android.job.Job
import com.evernote.android.job.JobCreator
import com.mifos.utils.Tags

/**
 * Created by aksh on 17/7/18.
 */
class OfflineJobCreator : JobCreator {
    override fun create(tag: String): Job? {
        return when (tag) {
            Tags.OfflineSyncCenter -> OfflineSyncCenter()
            Tags.OfflineSyncClient -> OfflineSyncClient()
            Tags.OfflineSyncGroup -> OfflineSyncGroup()
            Tags.OfflineSyncSavingsAccount -> OfflineSyncSavingsAccount()
            Tags.OfflineSyncLoanRepayment -> OfflineSyncLoanRepayment()
            else -> null
        }
    }
}